src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -243,3 +352,61 @@ void MatterDoorLockPluginServerInitCallback()
 }
 
 void MatterDoorLockClusterServerAttributeChangedCallback(const app::ConcreteAttributePath & attributePath) {}
+
+// =============================================================================
+// Pre-change callbacks for cluster attributes
+// =============================================================================
+
+chip::Protocols::InteractionModel::Status __attribute__((weak))
```

Comment:
All weak to allow custom implementations right ?

Reply:
Right. All callbacks that makes sense should be implemented on the app side.